pkg/controller/clusters/status.go

```diff
@@ -218,7 +230,13 @@ func updateWorkersReadyCondition(cluster *anywherev1.Cluster, machineDeployments
 	// so reflect that on the conditon with an appropriate message.
 	totalOutdated := totalReplicas - totalUpdatedReplicas
 	if totalOutdated > 0 {
-		conditions.MarkFalse(cluster, anywherev1.WorkersReadyCondition, anywherev1.RollingUpgradeInProgress, clusterv1.ConditionSeverityInfo, "Worker nodes not up-to-date yet, %d rolling (%d up to date)", totalReplicas, totalUpdatedReplicas)
+		upgradeReason := anywherev1.RollingUpgradeInProgress
+		if cluster.Spec.ControlPlaneConfiguration.UpgradeRolloutStrategy != nil {
```

Comment:
nit: can you add a comment saying we are using cp's upgrade strategy here because we already validate that all the machines have the same upgrade strategy.